binaries/workerserver/main.go

```diff
@@ -15,26 +17,18 @@ var httpPort = flag.Int("http_port", 9091, "port to serve http on")
 
 func main() {
 	flag.Parse()
-	go serveHTTP()
+	stat, _ := stats.NewCustomStatsReceiver(stats.NewFinagleStatsRegistry, 15*time.Second)
+	stat = stat.Precision(time.Millisecond)
+	endpoints.RegisterStats("/admin/metrics.json", stat)
+	endpoints.RegisterHealthCheck("/")
+	go endpoints.Serve(fmt.Sprintf(":%d", *httpPort))
 
 	protocolFactory := thrift.NewTBinaryProtocolFactoryDefault()
 	transportFactory := thrift.NewTTransportFactory()
 
-	err := server.Serve(server.NewHandler(), fmt.Sprintf(":%d", *thriftPort), transportFactory, protocolFactory)
+	handler := server.NewHandler(stat.Scope("workerserver"))
```

Comment:
You don't have to prefix stats with the service name, some services do some don't.  But just fyi absorber exports (service, source(host), metric name) so we already have this info.